### PR TITLE
fix(bindings): build on windows

### PR DESCRIPTION
**Pull request by OrangeSodahub**
- Update version of `fusefilter.h`.
- Fix `__umulh` not defined (MSVC on win-64bit).
- Fix error `C2540` (MSVC on win-64bit).
- Add ci/cd for windows.
- Fix unittest on windows.

### CODE REVIEW

**Comment by numb3r3** [comment]
@OrangeSodahub Please also enable window runner in `.github/workflows/ci.yml` and `.github/workflows/cd.yml`. And for more details, please refer to https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners

**Review by numb3r3 — CHANGES_REQUESTED** [review]
Please also enable window runner in .github/workflows/ci.yml and .github/workflows/cd.yml. And for more details, please refer to https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners

**Comment by OrangeSodahub** [comment]
Haven't found the correct way to use asterisk wildcard like linux on windows when run pytest yet. So put the specific file name in.

**Comment by OrangeSodahub** [comment]
Three problems:
- file name contains `:` is not allowed.
- Pod was forced to close after 1 second. Graceful closing is not available on Windows.
- UnicodeEncodeError

**Comment by numb3r3** [comment]
@OrangeSodahub About the UnicodeError, please refer to https://docs.jina.ai/get-started/install/windows/#unicodeencodeerror-on-jina-cli

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -63,8 +63,38 @@ jobs:
...
+          pip install --pre docarray
+          pip install -e ".[test]"
+      - name: Test
+        id: test
+        shell: pwsh
+        run: |
+          $env:PYTHONIOENCODING='utf-8'
+          cd tests/
+          pytest --suppress-no-test-exit-code --cov=annlite --cov-report=xml -v -s -m "not gpu" -k "test"
+          echo "::set-output name=codecov_flag::annlite"
```
Comment (numb3r3) [inline .github/workflows/cd.yml:93]: Please turn off the coverage report in the pytest on windows. And what's more, the target test cases should come from `${{ matrix.test-path }}`

```suggestion
          pytest -v -s -m "not gpu" ${{ matrix.test-path }}
```

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: .github/workflows/cd.yml (hunk 1 above)
Comment (OrangeSodahub) [reply .github/workflows/cd.yml:93]: `${{ matrix.test-path }}` contains `*`, however asterisk wildcard `*` is not supported in pytest on windows.

**Comment by OrangeSodahub** [comment]
On Windows, two adjacent calls of `datetime.datetime.utcnow()` could generate the two same time.

**Review by numb3r3 — APPROVED** [review]
LGTM 👍

**Pull request merged**